File: client-clojure/src/sirqul_io_t_platform/specs/retailer_short_response.clj
(ns sirqul-io-t-platform.specs.retailer-short-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.contact-info-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            )
  (:import (java.io File)))


(def retailer-short-response-data
  {
   (ds/opt :retailerId) int?
   (ds/opt :name) string?
   (ds/opt :facebookUrl) string?
   (ds/opt :twitterUrl) string?
   (ds/opt :active) boolean?
   (ds/opt :contact) contact-info-response-spec
   (ds/opt :logo) asset-short-response-spec
   (ds/opt :picture1) asset-short-response-spec
   (ds/opt :picture2) asset-short-response-spec
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :visibility) string?
   })

(def retailer-short-response-spec
  (ds/spec
    {:name ::retailer-short-response
     :spec retailer-short-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/RetailerShortResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import RetailerShortResponse._

case class RetailerShortResponse (
  retailerId: Option[Long],
name: Option[String],
facebookUrl: Option[String],
twitterUrl: Option[String],
active: Option[Boolean],
contact: Option[ContactInfoResponse],
logo: Option[AssetShortResponse],
picture1: Option[AssetShortResponse],
picture2: Option[AssetShortResponse],
latitude: Option[Double],
longitude: Option[Double],
visibility: Option[String])

object RetailerShortResponse {
  import DateTimeCodecs._

  implicit val RetailerShortResponseCodecJson: CodecJson[RetailerShortResponse] = CodecJson.derive[RetailerShortResponse]
  implicit val RetailerShortResponseDecoder: EntityDecoder[RetailerShortResponse] = jsonOf[RetailerShortResponse]
  implicit val RetailerShortResponseEncoder: EntityEncoder[RetailerShortResponse] = jsonEncoderOf[RetailerShortResponse]
}
